### PR TITLE
GLT-1000 Fix search on List Samples

### PR DESCRIPTION
**Pull request by callunity**

### CODE REVIEW

**Comment by apmasell** [comment]
:+1:

File: miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java [hunk 1]
```diff
@@ -1,6 +1,13 @@
...
 
-import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.*;
+import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.isAliquotSample;
+import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.isDetailedSample;
+import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.isIdentitySample;
+import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.isStockSample;
+import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.isStringBlankOrNull;
+import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.isStringEmptyOrNull;
+import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.isTissueProcessingSample;
+import static uk.ac.bbsrc.tgac.miso.core.util.LimsUtils.isTissueSample;
```
Comment (frankie13) [inline miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/impl/DefaultSampleService.java]: :100:

File: sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSampleDao.java [hunk 2]
```diff
@@ -346,21 +347,35 @@ public boolean aliasExists(String alias) throws IOException {
...
   @Override
+  public Long countBySearch(String querystr) throws IOException {
+    Criteria criteria = currentSession().createCriteria(SampleImpl.class);
+    criteria.add(Restrictions.or(
+        Restrictions.ilike("identificationBarcode", "%" + querystr + "%"),
+        Restrictions.ilike("name", "%" + querystr + "%"),
+        Restrictions.ilike("alias", "%" + querystr + "%")));
+    return (Long) criteria.setProjection(Projections.rowCount()).uniqueResult();
+  }
+
```
Comment (frankie13) [inline sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSampleDao.java]: Would :heart:  to see a test for this new method and perhaps an `@CoverageIgnore` for the trivial implementation in the service layer :neckbeard:

**Comment by frankie13** [comment]
RAD thank you! :) :+1:

**Pull request merged**